ggml-alloc.c

```diff
@@ -277,17 +310,68 @@ struct ggml_allocr * ggml_allocr_new(void * data, size_t size, size_t alignment)
     return alloc;
 }
 
-// address and size of the buffer when measuring
-// it needs to be large enough to fit all the tensors, but it cannot overlap with other existing buffers
-static void * const MEASURE_BASE_ADDR = (void *) 0x1000;
-static const size_t MEASURE_MAX_SIZE  = 1ULL<<40; // 1 TB
+// OS specific functions to allocate and free uncommitted virtual memory
+static void * alloc_vmem(size_t size) {
+#if defined(_WIN32)
+    return VirtualAlloc(NULL, size, MEM_RESERVE, PAGE_NOACCESS);
+#elif defined(_POSIX_MAPPED_FILES)
```

Comment:
If the emscripten build doesn't work, it can be excluded from using mmap here and in `free_vmem` by checking if `__EMSCRIPTEN__ ` is defined. I think this should do it:
```suggestion
#elif defined(_POSIX_MAPPED_FILES) && !defined(__EMSCRIPTEN__)
```

Reply:
Good news - the Emscripten build looks to be working without any adjustments needed